File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/common/tasks/MessageTask.java
package org.firstinspires.ftc.teamcode.common.tasks;

import org.firstinspires.ftc.teamcode.common.BunyipsOpMode;

public class MessageTask extends Task implements TaskImpl {

    private final String message;

    public MessageTask(BunyipsOpMode opMode, double time, String message) {
        super(opMode, time);
        this.message = message;
    }

    @Override
    public void run() {
        if (isFinished()) {
            return;
        }
        opMode.telemetry.addLine(String.format("%s || %.2f", message, time));
    }

}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/common/tasks/MessageTask.kt
package org.firstinspires.ftc.teamcode.common.tasks

import android.annotation.SuppressLint
import org.firstinspires.ftc.teamcode.common.BunyipsOpMode

/**
 * Relay a message to the driver station for a specific time.
 */
class MessageTask(opMode: BunyipsOpMode, time: Double, private val message: String) :
    Task(opMode, time), AutoTask {
    @SuppressLint("DefaultLocale")
    override fun run() {
        opMode.addTelemetry(String.format("%s || %.2f", message, time))
        opMode.telemetry.update()
    }

    override fun onFinish() {
        return
    }
}